themes/bootstrap3/templates/cart/cart.phtml

```diff
@@ -18,37 +18,45 @@
         </label>
       </div>
       <?php if ($this->userlist()->getMode() !== 'disabled'): ?>
-        <button type="submit" class="toolbar-btn btn-type-save" name="saveCart" value="1" title="<?=$this->transEscAttr('bookbag_save')?>" value="1">
+        <button type="submit" class="toolbar-btn btn-type-save" name="saveCart" title="<?=$this->transEscAttr('bookbag_save')?>" value="1">
           <?=$this->transEsc('Save')?>
         </button>
       <?php endif; ?>
-      <button type="submit" class="toolbar-btn btn-type-email" name="email" value="1" title="<?=$this->transEscAttr('bookbag_email')?>" value="1">
+      <button type="submit" class="toolbar-btn btn-type-email" name="email" title="<?=$this->transEscAttr('bookbag_email')?>" value="1">
         <?=$this->transEsc('Email')?>
       </button>
       <?php $exportOptions = $this->export()->getActiveFormats('bulk'); if (count($exportOptions) > 0): ?>
-        <button type="submit" class="toolbar-btn btn-type-export" name="export" value="1" title="<?=$this->transEscAttr('bookbag_export')?>" value="1">
+        <button type="submit" class="toolbar-btn btn-type-export" name="export" title="<?=$this->transEscAttr('bookbag_export')?>" value="1">
```

Comment:
I appreciate the catch on the double value, but why do we need value on these buttons? Is it just for validation?

Reply:
I believe that the value is needed by the cart processor -- it knows which action to execute based on which button is pressed and thus which value is received. See: https://github.com/vufind-org/vufind/blob/dev/module/VuFind/src/VuFind/Controller/CartController.php#L82-L96

Reply:
If I recall correctly, the cart knows which button was pressed thanks to this bit of Javascript: https://github.com/vufind-org/vufind/blob/dev/themes/bootstrap3/templates/cart/cart.phtml#L67-L70.

Maybe this is something we look to refactor in the future. There's probably a better way.

Reply:
The cart currently works with Javascript disabled. Whether we need to maintain that is a separate question, but the original design was meant to work without requiring Javascript.